plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -160,4 +165,54 @@ private Object toVarcharVarcharMapBlock(Map<String, String> values)
         blockBuilder.closeEntry();
         return varcharToVarcharMapType.getObject(blockBuilder, 0);
     }
+
+    private static VersionRange extractVersionRange(TupleDomain<Integer> constraint)
+    {
+        Domain versionDomain = constraint.getDomains()
+                .map(map -> map.get(VERSION_COLUMN_INDEX))
+                .orElse(Domain.all(BIGINT));
+        Optional<Long> startVersion = Optional.empty();
+        Optional<Long> endVersion = Optional.empty();
+        if (!(versionDomain.isAll() || versionDomain.isNone())) {
+            List<Range> orderedRanges = versionDomain.getValues().getRanges().getOrderedRanges();
+            if (orderedRanges.size() == 1) {
+                // Opt for a rather pragmatical choice of extracting the version range
+                // only when dealing with a single range
+                Range range = orderedRanges.get(0);
+                if (range.isSingleValue()) {
+                    long version = (long) range.getLowBoundedValue();
+                    startVersion = Optional.of(version);
+                    endVersion = Optional.of(version);
+                }
+                else {
+                    if (!range.isLowUnbounded()) {
+                        long version = (long) range.getLowBoundedValue();
+                        if (!range.isLowInclusive()) {
+                            version++;
+                        }
+                        startVersion = Optional.of(version);
+                    }
+                    if (!range.isHighUnbounded()) {
+                        long version = (long) range.getHighBoundedValue();
+                        if (!range.isHighInclusive()) {
+                            version--;
+                        }
+                        endVersion = Optional.of(version);
+                    }
+                }
+            }
+        }
+        return new VersionRange(startVersion, endVersion);
+    }
+
+    private record VersionRange(Optional<Long> startVersion, Optional<Long> endVersion)
+    {
+        @SuppressWarnings("UnusedVariable") // TODO: Remove once https://github.com/google/error-prone/issues/2713 is fixed
+        private VersionRange
+        {
+            requireNonNull(startVersion, "identity is null");
+            requireNonNull(endVersion, "identity is null");
+            verify(startVersion.orElse(0L) <= endVersion.orElse(startVersion.orElse(0L)), "startVersion is greater than endVersion");
```

Comment:
Looks like a duplicate code from DeltaLakeTransactionLogEntryIterator? we can extract into a method `verifyVersion`?